src/App.jsx

```diff
@@ -1,7 +1,24 @@
-export default function App() {
+import { AnimalListItem } from "./components/AnimalListItem";
+
+const App = () => {
+  const animals = [
+    "🐶 dog",
+    "🐈 cat",
+    "🐔 chicken",
+    "🐮 cow",
+    "🐏 sheep",
+    "🐴 horse",
+  ];
+
   return (
-    <div className="App">
-      <h1 className="text-2xl text-red-500">Hello React!</h1>
-    </div>
+    <main className="flex items-center justify-center h-screen">
+      <ul className="flex flex-col justify-center text-xl border rounded-md border-gray-300">
+        {animals.map((animal, index) => (
+          <AnimalListItem key={index}>{animal}</AnimalListItem>
```

Comment:
[https://legacy.reactjs.org/docs/lists-and-keys.html#:~:text=Keys%20should%20be%20given%20to,list%20item%20among%20its%20siblings.](url)

Refering to the react doc it is recommende to have key's value as a unique string

Reply:
@estherniyo thanks for the comment and also for providing the reference, I went through the link you shared and they recommend not using the index as keys when the order of the Array may change. But in the current case, it is certain that the component is static the order of the list never changes anyhow, don't you think it would be safe to use the index in this particular case?

Another option I was thinking of is using the list themselves as keys `(those 'dog', 'cats' values)` since they are unique and they are strings, But I'm not sure if it is a good idea, what do you think?

Also, I checked the new React documentation and in one of their examples ([linked here](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key)) they were using numbers as keys without converting them to strings. I don't think keys necessarily have to be strings

Reply:
However you can see from that second link that they are referring to their own defined id's not index for consistency


> Rules of keys 
> Keys must be unique among siblings. However, it’s okay to use the same keys for JSX nodes in different arrays.
> Keys must not change or that defeats their purpose! Don’t generate them while rendering.

consider those 2 lines in the document from the new react

Reply:
So you recommend having the ids predefined in the array itself